rockcraft/lifecycle.py

```diff
@@ -18,17 +18,41 @@
 """Lifecycle integration."""
 
 import datetime
+import os
 import subprocess
 from pathlib import Path
 
 from craft_cli import EmitterMode, emit
 
+from rockcraft.errors import RockcraftInitError
+
 from . import oci, providers, utils
 from .parts import PartsLifecycle, Step
 from .project import Project, load_project
 from .providers import capture_logs_from_instance
 
 
+def init(rockcraft_yaml_content: str) -> None:
+    """Initialize a rockcraft project.
+
+    :param rockcraft_yaml_content: Content of the rockcraft.yaml file
+    :type rockcraft_yaml_content: str
```

Comment:
What's everyone's opinion regarding `:type` directives in docstrings for functions with type annotations? Personally I think we can skip these

Reply:
I agree. I suggest we do this in another PR though, cause `:type` is in quite a few docstrings.

https://github.com/canonical/rockcraft/issues/58